src/session_impl.cpp

```diff
@@ -5636,14 +5662,21 @@ namespace aux {
 #ifndef TORRENT_NO_DEPRECATE
 	entry session_impl::dht_state() const
 	{
-		if (!m_dht) return entry();
-		return m_dht->state();
+		return m_dht ? dht::save_dht_state(m_dht->state()) : entry();
 	}
 
 	void session_impl::start_dht_deprecated(entry const& startup_state)
 	{
 		m_settings.set_bool(settings_pack::enable_dht, true);
-		start_dht(startup_state);
+		std::vector<char> tmp;
+		std::back_insert_iterator<std::vector<char>> out(tmp);
```

Comment:
instead of creating this ``out`` object, you can just say: ``bencode(std::back_inserter(tmp), startup_state);``